crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -103,15 +104,29 @@ impl Command for ConnectCommand {
         let bridge_config = bridge_config(config, &self.cloud).map_err(anyhow::Error::new)?;
         let credentials_path =
             credentials_path_for(config, &self.cloud).map_err(anyhow::Error::new)?;
-        let use_cryptoki = config.device.cryptoki.mode.enabled();
+
+        let cloud = config
+            .as_cloud_config((&self.cloud).into())
+            .map_err(anyhow::Error::new)?;
+
+        let cryptoki_key_uri = config.device.cryptoki_config(cloud)?.map(|c| match c {
+            CryptokiConfig::Direct(d) => d.uri,
+            CryptokiConfig::SocketService { uri, .. } => uri,
+        });
+        let cryptoki_mode = config.device.cryptoki.mode.clone();
+        let cryptoki_status = match cryptoki_key_uri {
+            None => "off".to_string(),
+            Some(None) => format!("{cryptoki_mode} (key: unspecified)"),
+            Some(Some(key)) => format!("{cryptoki_mode} (key: {key})"),
+        };
```

Comment:
This is working but I'm surprised by the `mode` extracted twice via two different paths (indirectly with `config.device.cryptoki_config` and then explicitly with `config.device.cryptoki.mode`). Isn't there a risk of inconsistency?

Reply:
I had to revert this because I forgot why I did that originally: `cloud.device.key_uri` is empty if we use `device.key_uri`, and `.cryptoki_config()` actually read both and applied `device.key_uri` if the one from `cloud` was `None`.